base/util.jl

```diff
@@ -301,8 +301,9 @@ end
 
 function with_output_color(f::Function, color::Union{Int, Symbol}, io::IO, args...)
     buf = IOBuffer()
+    buf_context = IOContext(io, buf)
```

Comment:
You are right. That was bad... Will add a test for `print_with_color` that would have caught this.